base/timing.jl

```diff
@@ -371,3 +371,37 @@ macro timed(ex)
         (value=val, time=elapsedtime/1e9, bytes=diff.allocd, gctime=diff.total_time/1e9, gcstats=diff)
     end
 end
+
+"""
+    @loadtime
+
+Enables the timing of any loading of packages and their dependencies within a given expression.
+
+```julia-repl
+julia> @loadtime using CSV
+      0.5 ms      IteratorInterfaceExtensions
+    325.1 ms    TableTraits
+    349.7 ms    SentinelArrays
+     82.5 ms    Parsers
+      0.6 ms    DataValueInterfaces
+      1.7 ms      DataAPI
+     19.7 ms    Tables
+     27.0 ms    PooledArrays
+    940.9 ms  CSV
+```
+
+!!! note
+    If a package's dependencies have already been loaded they will not appear under that package and the
+    package will accurately report a faster load time than if it were to be loaded in isolation.
+
+"""
+macro loadtime(ex)
```

Comment:
what about stupid people like me who write:

```

@loadtime begin
    # lots of indirection
    @loadtime begin 
       using ...
    end
   using A
end
```

Currently `A` won't be counted

Reply:
True. I guess instead of a bool it could be an Int that's incremented/decremented and checked for being > 0